### ISSUE TITLE
createReadStream() doesn't seem to be exposed

### ISSUE THREAD

**Issue by nraynaud** [Sullux/aws-sdk#11]
Hi, thanks for your nice wrapper.

Amongst my numerous normal cases, I have an odd case where I don't want to await `.promise()` but get a readstream on the `s3.getObject()` function, is there a way to do that?

Thanks,
Nicolas.

PS: I found you medium blog post by google, I was questioning my own sanity: "Am I crazy or none of my code is in the stacktrace, how I am supposed to debug that?"

**Comment by Sullux** [comment]
Hi Nicolas, good question! At the moment there's no way to get anything from the API response other than `.promise()` functionality. Since you've expressed interest, I'll take a look and see if there's something else I might be able to do. Maybe I could use `Proxy` to hang a differently-behaving function off the primary invocation function. Something along the lines of:

```javascript
const { s3 } = require('@sullux/aws-sdk')()
const stream = s3.getObject.raw({ Bucket: 'my-bucket', Key: 'myObject' }).createReadStream()
// profit
```

In the above pattern, the `raw` function would invoke the underlying `getObject` function but would skip the default `.promise()` behavior and give you the raw AWS response instead. Assuming I can make that work properly, would this solve your use case?

>  "Am I crazy or none of my code is in the stacktrace, how I am supposed to debug that?"

No, you're not crazy. The AWS SDK is gaslighting you. :smirk:

**Comment by nraynaud** [comment]
yes, thanks, I think exiting the pattern or even exposing the entire raw API (`s3.rawAws.getObject()` ) for the unforeseen situations is a good compromise, this is open source, we can't expect you to plan for everything I do.

**Comment by Sullux** [comment]
Ok, version `1.0.5` is officially published to NPM. I looked at the possibility of the `s3.raw.getObject()` pattern, but it was impractical because of the way the functions are late bound. I could do it, but it would have taken a whole lot more coding. Instead I implemented the raw version of the function as a property of the wrapping function as:

```javascript
const { s3 } = require('@sullux/aws-sdk')()
const stream = s3.getObject.raw({ Bucket: 'my-bucket', Key: 'myObject' }).createReadStream()
// profit
```

I tested it a bit and it seems to behave as expected. Let me know if you have any problems getting it to work. I'll leave this thread open for a few days in case we need follow up.
